# EXPLORACION Y LIMPIEZA DE LOS DATASETS 

In [1]:
import pandas as pd
import numpy as np
pd.set_option('display.max_colwidth', None)
pd.set_option('display.max_rows',None)
pd.set_option('display.max_columns',None)

Recuperamos los datasets guardados en formato CSV :

In [2]:
df_Alquiler=pd.read_csv('df_Alquiler.csv')
df_Comprar=pd.read_csv('df_Comprar.csv')
Alquiler_Parking=pd.read_csv('Alquiler_Parking.csv')
Comprar_Parking=pd.read_csv('Comprar_Parking.csv')
Alquiler_Amueblado=pd.read_csv('Alquiler_Amueblado.csv')

## CREACION DEL DATASET DE VIVIENDAS EN ALQUILER

## Creamos un 'id' único

Partimos de la base que la variable 'id' que representa cada anuncio debería ser única :

In [3]:
df_Alquiler.shape

(4486, 16)

In [4]:
df_Alquiler.nunique()

Unnamed: 0     4486
id             4473
url            4486
subtype          11
transaction       1
owner             2
sell-type         1
provider        721
title          3728
location         68
features       3350
description    3649
price           959
price2          521
m2              293
rooms            16
dtype: int64

De un total de 4486, solo tenemos 4473 id únicos de manera que 13 son duplicados :

In [5]:
df_Alquiler[df_Alquiler.duplicated(subset='id',keep=False)]

Unnamed: 0                id  \
644          644  id30466000039130   
645          645  id30466000039130   
1740        1740  id13074000029974   
1741        1741  id13074000029864   
1742        1742  id13074000029871   
1744        1744  id13074000029871   
1746        1746  id13074000029864   
1748        1748  id13074000029974   
2192        2192  id27758000002133   
2193        2193  id27758000002133   
3812        3812   id8166003789851   
3813        3813  id14453004331759   
3815        3815   id8166003789851   
3818        3818  id14453004331759   
3948        3948   id4737003814477   
3950        3950   id4737003814477   
4068        4068   id5714000000326   
4069        4069   id5714000000326   
4171        4171  id21241000011690   
4172        4172   id4017003409219   
4173        4173   id4017003409566   
4174        4174   id4017003409566   
4175        4175   id4017003409219   
4177        4177  id21241000011690   
4262        4262   id5714000000443   
4264        4264   id5714000000443   

                                                                                                                                                                                                           url  \
644                      https://www.habitaclia.com/alquiler-apartamento-de_dos_habitaciones_situado_en_el_corazon_de_barcelo_raval-barcelona-i30466000039130.htm?pag=42&f=&geo=d&coddists=100&from=list&lo=55   
645                      https://www.habitaclia.com/alquiler-apartamento-de_dos_habitaciones_situado_en_el_corazon_de_barcelo_raval-barcelona-i30466000039130.htm?pag=43&f=&geo=d&coddists=100&from=list&lo=55   
1740            https://www.habitaclia.com/alquiler-piso-esplendido_de_temporada_de_1_a_11_meses_en_l_antiga_esquerra_de_l_eixample-barcelona-i13074000029974.htm?pag=53&f=&geo=d&coddists=200&from=list&lo=55   
1741      https://www.habitaclia.com/alquiler-piso-magnifico_de_temporada_de_1_a_11_meses_en_dreta_de_leixample_dreta_de_l_eixample-barcelona-i13074000029864.htm?pag=53&f=&geo=d&coddists=200&from=list&lo=55   
1742                        https://www.habitaclia.com/alquiler-piso-de_calidad_de_temporada_de_1_a_11_meses_en_dreta_de_l_eixample-barcelona-i13074000029871.htm?pag=53&f=&geo=d&coddists=200&from=list&lo=55   
1744                        https://www.habitaclia.com/alquiler-piso-de_calidad_de_temporada_de_1_a_11_meses_en_dreta_de_l_eixample-barcelona-i13074000029871.htm?pag=54&f=&geo=d&coddists=200&from=list&lo=55   
1746      https://www.habitaclia.com/alquiler-piso-magnifico_de_temporada_de_1_a_11_meses_en_dreta_de_leixample_dreta_de_l_eixample-barcelona-i13074000029864.htm?pag=54&f=&geo=d&coddists=200&from=list&lo=55   
1748            https://www.habitaclia.com/alquiler-piso-esplendido_de_temporada_de_1_a_11_meses_en_l_antiga_esquerra_de_l_eixample-barcelona-i13074000029974.htm?pag=54&f=&geo=d&coddists=200&from=list&lo=55   
2192                 https://www.habitaclia.com/alquiler-piso-precioso_apartamento_reformado_en_barrio_vallcarca_vallcarca_penitents-barcelona-i27758000002133.htm?pag=5&f=&geo=d&coddists=600&from=list&lo=55   
2193                 https://www.habitaclia.com/alquiler-piso-precioso_apartamento_reformado_en_barrio_vallcarca_vallcarca_penitents-barcelona-i27758000002133.htm?pag=6&f=&geo=d&coddists=600&from=list&lo=55   
3812                                         https://www.habitaclia.com/alquiler-piso-exclusivo_en_sant_gervasi_sant_gervasi_bonanova-barcelona-i8166003789851.htm?pag=8&f=&geo=d&coddists=500&from=list&lo=55   
3813        https://www.habitaclia.com/alquiler-casa-exclusiva_con_piscina_de_40_m2_jardin_y_cinco_terrazas_en_sant_gervasi_bonanova-barcelona-i14453004331759.htm?pag=8&f=&geo=d&coddists=500&from=list&lo=55   
3815                                         https://www.habitaclia.com/alquiler-piso-exclusivo_en_sant_gervasi_sant_gervasi_bonanova-barcelona-i8166003789851.htm?pag=9&f=&geo=d&coddists=500&from=list&lo=55   
3818        https://www.habitaclia.com/

Al constatar que se tratan de anuncios repetidos, eliminamos duplicados y nos quedamos con la primera ocurrencia :

In [6]:
df_Alquiler.drop_duplicates(subset=['id'], keep='first',inplace=True)

In [7]:
df_Alquiler.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 4473 entries, 0 to 4485
Data columns (total 16 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   Unnamed: 0   4473 non-null   int64 
 1   id           4473 non-null   object
 2   url          4473 non-null   object
 3   subtype      4473 non-null   object
 4   transaction  4473 non-null   object
 5   owner        4473 non-null   object
 6   sell-type    4473 non-null   object
 7   provider     4473 non-null   object
 8   title        4473 non-null   object
 9   location     4473 non-null   object
 10  features     4473 non-null   object
 11  description  4473 non-null   object
 12  price        4469 non-null   object
 13  price2       4473 non-null   int64 
 14  m2           4473 non-null   int64 
 15  rooms        4473 non-null   int64 
dtypes: int64(4), object(12)
memory usage: 594.1+ KB


In [8]:
Alquiler_Parking.shape

(436, 15)

In [9]:
Alquiler_Parking.nunique()

id             436
url            436
subtype         11
transaction      1
owner            2
sell-type        1
provider       159
title          415
location        48
features       397
description    413
price          195
price2         140
m2             190
rooms           11
dtype: int64

In [10]:
Alquiler_Amueblado.shape

(2588, 15)

In [11]:
Alquiler_Amueblado.nunique()

id             2437
url            2588
subtype          11
transaction       1
owner             2
sell-type         1
provider        332
title          1926
location         63
features       1895
description    2135
price           568
price2          292
m2              223
rooms            10
dtype: int64

In [12]:
Alquiler_Amueblado[Alquiler_Amueblado.duplicated(subset='id',keep=False)]

id  \
221   id31390000000182   
222   id31390000000243   
225   id31390000000243   
226   id31390000000182   
314    id4975004332821   
315    id4975004332821   
329    id4737003817086   
331    id4737003817086   
365    id4737003817072   
368    id8166003792228   
374    id4737003816863   
375    id4737003816863   
377    id8166003792228   
379    id4737003817072   
402    id1507004169374   
403   id27758000001983   
407   id27758000001983   
413    id1507004169374   
432   id31390000000096   
433   id27758000002016   
434    id8166003793236   
435   id27758000002016   
436    id8166003793236   
439   id31390000000096   
449    id8166003792537   
451    id8166003792537   
463    id3317002935390   
464    id9873002102985   
468    id9873002102985   
469    id3317002935390   
479    id8166003790680   
480    id8166003790680   
492    id8166003792381   
493   id31390000000098   
494   id31390000000101   
495    id8166003792381   
499   id31390000000101   
500   id31390000000098   
507    id1921002848347   
510    id1921002848347   
539    id8166003793046   
541    id8166003793046   
644    id5299004351003   
645    id5299004351003   
674    id2241000000799   
682    id2241000000799   
689   id29057000000050   
691   id29057000000050   
703    id2277003487137   
704   id27758000002107   
709   id27758000002107   
713    id2277003487137   
777    id4737003816637   
780    id4737003816637   
791   id21241000011431   
797   id21241000011431   
809   id27758000001925   
810   id27758000001925   
834   id31390000000172   
837   id27758000001790   
839   id27758000002092   
840   id31390000000172   
841   id27758000001790   
842   id27758000002092   
846   id27758000001556   
854   id38623000000487   
855   id27758000001556   
857   id38623000000487   
880     id531205770962   
881   id27758000001148   
882    id5714000000303   
884    id4027004298733   
885    id4027004298733   
886     id531205770962   
887   id27758000001148   
890    id5714000000303   
944   id27758000001979   
946   id27758000001979   
959   id27758000002097   
962   id27758000002097   
1003  id21241000011452   
1004  id21241000011640   
1006  id21241000011452   
1008  id21241000011640   
1015  id27758000001832   
1016  id27758000001573   
1020  id27758000001573   
1021  id27758000001832   
1031  id21241000011547   
1034   id5714000000270   
1035  id21241000011547   
1036   id5714000000270   
1047   id9295003949793   
1050   id9295003949793   
1062   id5714000000249   
1063   id1507004169362   
1065   id1507004169362   
1066   id5714000000249   
1074  id27758000001157   
1075   id1507004169353   
1076  id27758000002090   
1078  id38623000000447   
1079  id43080000000028   
1080  id27758000002090   
1082   id1507004169353   
1086  id27758000001157   
1087  id38623000000447   
1088  id43080000000028   
1094   id8166003792847   
1095   id8166003792847   
1119  id14760003330099   
1120   id5714000000574   
1122  id11941003895327   
1123  id30785000001900   
1125  id30785000001900   
1126   id5714000000574   
1127  id11941003895327   
1128  id14760003330099   
1145  id44432000000054   
1147  id44432000000044   
1148  id44432000000046   
1149  id44432000000042   
1150  id44432000000079   
1153  id44432000000051   
1159  id44432000000044   
1160  id44432000000051   
1161  id44432000000046   
1162  id44432000000079   
1163  id44432000000054   
1164  id44432000000042   
1169   id8166003790412   
1170   id8166003790412   
1210  id44432000000063   
1211  id44432000000081   
1216  id44432000000063   
1217  id44432000000081   
1244   id5714000000436   
1247   id5714000000436   
1273  id14760003329922   
1274   id3839004360477   
1275   id3839004360477   
1276  id14760003329922   
1288  id13074000029246   
1289  id13074000029242   
1292  id13074000029242   
1302  id13074000029246   
1305  id13074000029246   
1331   id4840004296869   
1332   id5714000000301   
1336   id4840004296869   
1337   id5714000000301   
1379  id13074000024176   
1381  id13074000024176   
1409   id5714000

In [13]:
Alquiler_Amueblado.drop_duplicates(subset=['id'], keep='first',inplace=True)

In [14]:
Alquiler_Amueblado.nunique()

id             2437
url            2437
subtype          11
transaction       1
owner             2
sell-type         1
provider        332
title          1926
location         63
features       1895
description    2135
price           568
price2          292
m2              223
rooms            10
dtype: int64

## Incorporamos variables amueblado Si/No y parking Si/No

Usamos la función assign() para incorporar las variables amueblado y parking en el dataset :

In [15]:
df_Alquiler=df_Alquiler.assign(amueblado=df_Alquiler.id.isin(Alquiler_Amueblado.id).astype(int))

In [16]:
df_Alquiler['amueblado']=np.where(df_Alquiler['amueblado']==1,'Si','No')

In [17]:
df_Alquiler['amueblado'].value_counts()

Si    2281
No    2192
Name: amueblado, dtype: int64

In [18]:
df_Alquiler=df_Alquiler.assign(parking=df_Alquiler.id.isin(Alquiler_Parking.id).astype(int))

In [19]:
df_Alquiler['parking']=np.where(df_Alquiler['parking']==1,'Si','No')

In [20]:
df_Alquiler['parking'].value_counts()

No    4042
Si     431
Name: parking, dtype: int64

In [21]:
df_Alquiler.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 4473 entries, 0 to 4485
Data columns (total 18 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   Unnamed: 0   4473 non-null   int64 
 1   id           4473 non-null   object
 2   url          4473 non-null   object
 3   subtype      4473 non-null   object
 4   transaction  4473 non-null   object
 5   owner        4473 non-null   object
 6   sell-type    4473 non-null   object
 7   provider     4473 non-null   object
 8   title        4473 non-null   object
 9   location     4473 non-null   object
 10  features     4473 non-null   object
 11  description  4473 non-null   object
 12  price        4469 non-null   object
 13  price2       4473 non-null   int64 
 14  m2           4473 non-null   int64 
 15  rooms        4473 non-null   int64 
 16  amueblado    4473 non-null   object
 17  parking      4473 non-null   object
dtypes: int64(4), object(14)
memory usage: 664.0+ KB


In [22]:
df_Alquiler.head()

Unnamed: 0                id  \
0           0   id2277003487234   
1           1   id3974003524929   
2           2   id1948004073754   
3           3  id27758000002114   
4           4   id4737003817059   

                                                                                                                                                                             url  \
0  https://www.habitaclia.com/alquiler-piso-de_100m2_con_dos_habitaciones_totalmente_amueblado_en_plaza_gotic-barcelona-i2277003487234.htm?f=&geo=d&coddists=100&from=list&lo=55   
1                                                              https://www.habitaclia.com/alquiler-piso-gotic-barcelona-i3974003524929.htm?f=&geo=d&coddists=100&from=list&lo=55   
2   https://www.habitaclia.com/alquiler-piso-con_2_habitaciones_amueblado_con_ascensor_calefaccion_y_ai_gotic-barcelona-i1948004073754.htm?f=&geo=d&coddists=100&from=list&lo=55   
3    https://www.habitaclia.com/alquiler-piso-de_temporal_de_1_habitacion_a_unos_pasos_del_mar_y_barceloneta-barcelona-i27758000002114.htm?f=&geo=d&coddists=100&from=list&lo=55   
4       https://www.habitaclia.com/alquiler-piso-planta_baja_amueblada_de_tres_dormitorios_en_el_gotico_gotic-barcelona-i4737003817059.htm?f=&geo=d&coddists=100&from=list&lo=55   

  subtype transaction         owner    sell-type provider  \
0    FLAT        RENT  PROFESSIONAL  SECOND_HAND   2277-2   
1    FLAT        RENT  PROFESSIONAL  SECOND_HAND   3974-1   
2    FLAT        RENT  PROFESSIONAL  SECOND_HAND   1948-1   
3    FLAT        RENT  PROFESSIONAL  SECOND_HAND  27758-1   
4    FLAT        RENT  PROFESSIONAL  SECOND_HAND   4737-1   

                                                                                                      title  \
0              Alquiler Piso Plaza del pí. Piso de 100m² con dos habitaciones totalmente amueblado en plaza   
1                                                                          Alquiler Piso Carrer de la mercè   
2                  Alquiler Piso en Gòtic. Piso con 2 habitaciones amueblado con ascensor, calefacción y ai   
3  Alquiler Piso Passeig de joan de borbó. Piso de alquiler temporal de 1 habitación a unos pasos del mar y   
4                            Alquiler Piso en Gòtic. Planta baja amueblada de tres dormitorios en el gótico   

                  location                                      features  \
0        Barcelona - Gòtic  103m2 - 2 habitaciones - 2 baños - 19,42€/m2   
1        Barcelona - Gòtic   95m2 - 2 habitaciones - 2 baños - 20,00€/m2   
2        Barcelona - Gòtic    63m2 - 2 habitaciones - 1 baño - 33,33€/m2   
3  Barcelona - Barceloneta      52m2 - 1 habitacion - 1 baño - 33,65€/m2   
4        Barcelona - Gòtic   150m2 - 3 habitaciones - 1 baño - 13,33€/m2   

                                                                                                                   description  \
0    Piso de 100m² en Plaza del Pí con acceso a terraza comunitaria. La vivienda dispone de un amplio salón comedor exterio...   
1  EXCLUSIVIDAD - FINCA NUEVA - TERRAZA - PISCINA - WIFI - ALTO STANDING- Piso de 2 habitaciones en finca totalmente rehabi...   
2  Este apartamento cómodo de buenas dimensiones tiene todo lo que necesita para vivir en el casco antiguo de Barcelona. Ub...   
3      Apartamento de alquiler temporal (desde 32 días a 11 meses) Disponibilidad inmediata Este moderno piso de 52 m2 tien...   
4   Situado en una finca regia de El Gótico, cerca de la Plaza Real y La Rambla, se ubica esta planta baja amueblada de 150...   

                                   price  price2   m2  rooms amueblado parking  
0  Oportunidad\n2.000 €\nha bajado 100 €    2000  103      2        No      No  
1               1.900 €\nha bajado 100 €    1900   95      2        Si      No  
2                                2.100 €    2100   63      2        Si      No  
3                                1.750 €    1750   52      1        No      No  
4                                2

## Analizamos los valores null

In [23]:
df_Alquiler.isnull().sum()

Unnamed: 0     0
id             0
url            0
subtype        0
transaction    0
owner          0
sell-type      0
provider       0
title          0
location       0
features       0
description    0
price          4
price2         0
m2             0
rooms          0
amueblado      0
parking        0
dtype: int64

In [24]:
df_Alquiler[df_Alquiler['price'].isnull()]

Unnamed: 0               id  \
2127        2127  id1404004439701   
4480        4480  id8166003790272   
4481        4481  id4975004333325   
4482        4482  id4737003817044   

                                                                                                                                                                                                                       url  \
2127                                                                   https://www.habitaclia.com/alquiler-piso-obra_nueva_en_vallcarca_penitents-barcelona-i1404004439701.htm?pag=1&f=&geo=d&coddists=600&from=list&lo=55   
4480                              https://www.habitaclia.com/alquiler-chalet-villa_senorial_con_gran_jardin_junto_a_teknon_sant_gervasi_bonanova-barcelona-i8166003790272.htm?pag=53&f=&geo=d&coddists=500&from=list&lo=55   
4481  https://www.habitaclia.com/alquiler-chalet-fantastica_villa_de_diseno_de_10_dormitorios_incluida_la_habita_vallvidrera_tibidabo_les_planes-barcelona-i4975004333325.htm?pag=53&f=&geo=d&coddists=500&from=list&lo=55   
4482                                     https://www.habitaclia.com/alquiler-piso-casa_en_de_lujo_y_obra_nueva_con_seis_habitaciones_en_c_sarria-barcelona-i4737003817044.htm?pag=53&f=&geo=d&coddists=500&from=list&lo=55   

     subtype transaction         owner    sell-type provider  \
2127    FLAT        RENT  PROFESSIONAL  SECOND_HAND   1404-1   
4480   HOUSE        RENT  PROFESSIONAL  SECOND_HAND   8166-1   
4481   HOUSE        RENT  PROFESSIONAL  SECOND_HAND   4975-1   
4482    FLAT        RENT  PROFESSIONAL  SECOND_HAND   4737-1   

                                                                                                                         title  \
2127                                                      Alquiler Piso en Vallcarca - Penitents. Obra nueva en alquiler!!!!!!   
4480                                                     Alquiler Chalet Vilana. Villa señorial con gran jardín junto a teknon   
4481  Alquiler Chalet en Vallvidrera - Tibidabo - Les Planes. Fantástica villa de diseño de 10 dormitorios ,incluída la habita   
4482                                 Alquiler Piso en Sarrià. Casa en alquiler de lujo y obra nueva con seis habitaciones en c   

                                             location  \
2127                Barcelona - Vallcarca - Penitents   
4480              Barcelona - Sant Gervasi - Bonanova   
4481  Barcelona - Vallvidrera - Tibidabo - Les Planes   
4482                               Barcelona - Sarrià   

                               features  \
2127   110m2 - 3 habitaciones - 2 baños   
4480  450m2 - 12 habitaciones - 6 baños   
4481  650m2 - 10 habitaciones - 6 baños   
4482   395m2 - 6 habitaciones - 7 baños   

                                                                                                                      description  \
2127    Magníficos pisos a Estrenar, con magnifica terraza de 40 m2 con acceso privado . Dúplex con balcón más terraza de 60m2...   
4480  Esta casa y su encantador jardín son una pieza única en la Bonanova. Se compone de 3 plantas, en total 450 metros constr...   
4481  Esta magnífica villa de lujo de 650 m² se asienta sobre una parcela de 1.000 m². Destaca por su diseño especial y la amp...   
4482  Espectacular y moderna casa de 400m² con piscina privada y vistas sensacionales de Barcelona. Se encuentra en la zona al...   

     price  price2   m2  rooms amueblado parking  
2127   NaN       0  110      3        No      No  
4480   NaN       0  450     12        No      No  
4481   NaN       0  650     10        Si      Si  
4482   NaN       0  395      6        Si      Si

Al no tener información sobre el precio, anulamos estos anuncios :

In [25]:
df_Alquiler.drop(df_Alquiler[df_Alquiler['price'].isnull()].index, inplace = True)

In [26]:
df_Alquiler.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 4469 entries, 0 to 4485
Data columns (total 18 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   Unnamed: 0   4469 non-null   int64 
 1   id           4469 non-null   object
 2   url          4469 non-null   object
 3   subtype      4469 non-null   object
 4   transaction  4469 non-null   object
 5   owner        4469 non-null   object
 6   sell-type    4469 non-null   object
 7   provider     4469 non-null   object
 8   title        4469 non-null   object
 9   location     4469 non-null   object
 10  features     4469 non-null   object
 11  description  4469 non-null   object
 12  price        4469 non-null   object
 13  price2       4469 non-null   int64 
 14  m2           4469 non-null   int64 
 15  rooms        4469 non-null   int64 
 16  amueblado    4469 non-null   object
 17  parking      4469 non-null   object
dtypes: int64(4), object(14)
memory usage: 663.4+ KB


In [27]:
df_Alquiler.isna().sum()

Unnamed: 0     0
id             0
url            0
subtype        0
transaction    0
owner          0
sell-type      0
provider       0
title          0
location       0
features       0
description    0
price          0
price2         0
m2             0
rooms          0
amueblado      0
parking        0
dtype: int64

## Analizamos las viviendas duplicadas

Es posible que una vivienda sea anunciada por varias inmobiliarias, o que incluso una misma inmobiliaria anuncie la misma vivienda en anuncios diferentes :

In [28]:
df_Alquiler[df_Alquiler.duplicated(subset=['title','location', 'features', 'price', 'price2', 'm2',
       'rooms'],keep=False)].shape

(461, 18)

In [29]:
df_Alquiler[df_Alquiler.duplicated(subset=['location', 'features', 'price', 'price2', 'm2',
       'rooms'],keep=False)].shape

(1043, 18)

Un análisis del dataset nos lleva a concluir que en la mayoría de los casos, cuando una vivienda comparte el barrio y las carácterísticas principales (m2, precio, nºhabitaciones, nºbaños) de otra vivienda, se trata en realidad de la misma vivienda, de manera que decidimos eliminar los duplicados y quedarnos con la primera ocurrencia :

In [30]:
df_Alquiler.drop_duplicates(subset=['location', 'features', 'price', 'price2', 'm2',
       'rooms'], keep='first',inplace=True)

In [31]:
df_Alquiler.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 3814 entries, 0 to 4484
Data columns (total 18 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   Unnamed: 0   3814 non-null   int64 
 1   id           3814 non-null   object
 2   url          3814 non-null   object
 3   subtype      3814 non-null   object
 4   transaction  3814 non-null   object
 5   owner        3814 non-null   object
 6   sell-type    3814 non-null   object
 7   provider     3814 non-null   object
 8   title        3814 non-null   object
 9   location     3814 non-null   object
 10  features     3814 non-null   object
 11  description  3814 non-null   object
 12  price        3814 non-null   object
 13  price2       3814 non-null   int64 
 14  m2           3814 non-null   int64 
 15  rooms        3814 non-null   int64 
 16  amueblado    3814 non-null   object
 17  parking      3814 non-null   object
dtypes: int64(4), object(14)
memory usage: 566.1+ KB


## Creación de nuevas variables

A partir de los datos disponibles en el dataset, nos parece interesante extraer y/o añadir la siguiente información :
- distrito : la variable 'location' representa los barrios de la ciudad, nos interesa crear la variable 'distrito' que incorpora los 10 distritos de la ciudad, ya que muy probablemente la variable 'location' con 74 ocurrencias no sea de aplicación para nuestros modelos de machine learning.
- temporalidad : la modalidad de alquiler temporal, inferior a 11 meses, con toda seguridad tiene una influencia en el precio, nos planteamos extraer esta información de la variable 'description' del anuncio.
- cambio de precio : es evidente que el hecho de que una vivienda haya bajado de precio tiene un impacto en su precio final, nos planteamos extraer esta información de la variable 'precio'.
- nº de baños : esta información no existe como variable numérica, nos planteamos crearla a partir de la variable 'features'.

### Variable distrito

In [32]:
df_Comprar['location'].nunique()

74

In [33]:
barris=pd.DataFrame(df_Comprar['location'].value_counts())
barris

location
Barcelona - Dreta de l´Eixample                               1124
Barcelona - Raval                                             1092
Barcelona - Gòtic                                              747
Barcelona - Sagrada Família                                    718
Barcelona - La Nova Esquerra de l´Eixample                     687
Barcelona - Sant Gervasi - Galvany                             646
Barcelona - L´Antiga Esquerra de l´Eixample                    633
Barcelona - St. Pere - Sta. Caterina - El Born                 566
Barcelona - El Poble Sec - Parc de Montjuïc                    507
Barcelona - Sants-Badal                                        465
Barcelona - Sant Andreu de Palomar                             461
Barcelona - El Camp d´en Grassot i Gràcia Nova                 444
Barcelona - Sant Antoni                                        417
Barcelona - Vilapicina - Torre Llobeta                         403
Barcelona - Vila de Gràcia                                     397
Barcelona - Sants                                              372
Barcelona - Guinardó                                           349
Barcelona - La Marina del Port                                 348
Barcelona - El Camp de l´Arpa del Clot                         340
Barcelona - La Verneda - La Pau                                335
Barcelona - Hostafrancs                                        307
Barcelona - Barri de les Corts                                 307
Barcelona - Sant Martí de Provençals                           302
Barcelona - Diagonal Mar i el Front Marítim del Poblenou       294
Barcelona - Roquetes                                           279
Barcelona - Besòs - Maresme                                    262
Barcelona - Putget - Farró                                     258
Barcelona - Carmel                                             257
Barcelona - Pedralbes                                          256
Barcelona - Sarrià                                             244
Barcelona - Poblenou                                           241
Barcelona - Baix Guinardó                                      231
Barcelona - Trinitat Vella                                     226
Barcelona - La Bordeta                                         220
Barcelona - Sant Gervasi - Bonanova                            217
Barcelona - La Teixonera                                       215
Barcelona - Turó de la Peira                                   212
Barcelona - La Sagrera                                         211
Barcelona - Sant Ramon - Maternitat                            187
Barcelona - Fort Pienc                                         179
Barcelona - Prosperitat                                        170
Barcelona - Provençals del Poblenou                            163
Barcelona - Congrés - Indians                                  163
Barcelona - Tres Torres                                        158
Barcelona - Porta                                              155
Barcelona - El Clot                                            149
Barcelona - Horta                                              139
Barcelona - Barceloneta                                        137
Barcelona - Verdum                                             128
Barcelona - Ciutat Meridiana                                   124
Barcelona - El Parc i la Llacuna del Poblenou                  118
Barcelona - Can Baró                                           112
Barcelona - Vallcarca - Penitents                              111
Barcelona - Vallvidrera - Tibidabo - Les Planes                110
Barcelona - La Salut                                           108
Barcelona - Font de la Guatlla                                  95
Barcelona - Navas                                               87
Barcelona - La Font d´en Fargues                                83
Barcelona - Bon Pastor                                          82
Barcelona - Sant Genís dels Agudells  

In [34]:
barris.to_excel('barris.xlsx')

Recuperamos en excel el fichero barris.xlsx, creamos una columna con el nombre del distrito de cada barrio y creamos el fichero correspondiente distritos.xlsx que volvemos a importar :

In [35]:
distritos=pd.read_excel('distritos.xlsx')
distritos

location  \
0                                  Barcelona - Baix Guinardó   
1                                    Barcelona - Barceloneta   
2                                  Barcelona - Baró de Viver   
3                             Barcelona - Barri de les Corts   
4                                Barcelona - Besòs - Maresme   
5                                     Barcelona - Bon Pastor   
6                                       Barcelona - Can Baró   
7                                    Barcelona - Can Peguera   
8                                      Barcelona - Canyelles   
9                                         Barcelona - Carmel   
10                              Barcelona - Ciutat Meridiana   
11                             Barcelona - Congrés - Indians   
12  Barcelona - Diagonal Mar i el Front Marítim del Poblenou   
13                           Barcelona - Dreta de l´Eixample   
14            Barcelona - El Camp d´en Grassot i Gràcia Nova   
15                    Barcelona - El Camp de l´Arpa del Clot   
16                                       Barcelona - El Clot   
17                                       Barcelona - El Coll   
18             Barcelona - El Parc i la Llacuna del Poblenou   
19               Barcelona - El Poble Sec - Parc de Montjuïc   
20                            Barcelona - Font de la Guatlla   
21                                    Barcelona - Fort Pienc   
22                                         Barcelona - Gòtic   
23                                      Barcelona - Guinardó   
24                                         Barcelona - Horta   
25                                   Barcelona - Hostafrancs   
26               Barcelona - L´Antiga Esquerra de l´Eixample   
27                                    Barcelona - La Bordeta   
28                                      Barcelona - La Clota   
29                          Barcelona - La Font d´en Fargues   
30                                  Barcelona - La Guineueta   
31                            Barcelona - La Marina del Port   
32                    Barcelona - La Marina del Prat Vermell   
33                Barcelona - La Nova Esquerra de l´Eixample   
34                                    Barcelona - La Sagrera   
35                                      Barcelona - La Salut   
36                                  Barcelona - La Teixonera   
37                           Barcelona - La Verneda - La Pau   
38                 Barcelona - La Vila Olímpica del Poblenou   
39                                       Barcelona - Montbau   
40                                         Barcelona - Navas   
41                                     Barcelona - Pedralbes   
42                                      Barcelona - Poblenou   
43                                         Barcelona - Porta   
44                                   Barcelona - Prosperitat   
45                       Barcelona - Provençals del Poblenou   
46                                Barcelona - Putget - Farró   
47                                         Barcelona - Raval   
48                                      Barcelona - Roquetes   
49                               Barcelona - Sagrada Família   
50                        Barcelona - Sant Andreu de Palomar   
51                                   Barcelona - Sant Antoni   
52                      Barcelona - Sant Genís dels Agudells   
53                       Barcelona - Sant Gervasi - Bonanova   
54                        Barcelona - Sant Gervasi - Galvany   
55                      Barcelona - Sant Martí de Provençals   
56                       Barcelona - Sant Ramon - Maternitat   
57                                         Barcelona - Sants   
58                                   Barcelona - Sants-Badal   
59                                        Barcelona - Sarrià   
60            Barcelona - St. Pere - Sta. Caterina - El Born   
61                                    Barcelona - Torre Baró   
62                  

In [36]:
distritos.distrito.value_counts()

Nou Barris             13
Horta Guinardo         11
Sant Marti             10
Sants Montjuic          9
Sant Andreu             7
Eixample                6
Sarria Sant Gervasi     6
Gracia                  5
Ciutat Vella            4
Les Corts               3
Name: distrito, dtype: int64

In [37]:
df_Alquiler=pd.merge(df_Alquiler,distritos[['distrito','location']],on='location',how='left')

In [38]:
df_Alquiler.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 3814 entries, 0 to 3813
Data columns (total 19 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   Unnamed: 0   3814 non-null   int64 
 1   id           3814 non-null   object
 2   url          3814 non-null   object
 3   subtype      3814 non-null   object
 4   transaction  3814 non-null   object
 5   owner        3814 non-null   object
 6   sell-type    3814 non-null   object
 7   provider     3814 non-null   object
 8   title        3814 non-null   object
 9   location     3814 non-null   object
 10  features     3814 non-null   object
 11  description  3814 non-null   object
 12  price        3814 non-null   object
 13  price2       3814 non-null   int64 
 14  m2           3814 non-null   int64 
 15  rooms        3814 non-null   int64 
 16  amueblado    3814 non-null   object
 17  parking      3814 non-null   object
 18  distrito     3814 non-null   object
dtypes: int64(4), object(15)
mem

In [39]:
df_Alquiler.distrito.value_counts()

Eixample               1048
Ciutat Vella            755
Sarria Sant Gervasi     648
Gracia                  310
Sant Marti              309
Sants Montjuic          248
Les Corts               198
Horta Guinardo          138
Sant Andreu              81
Nou Barris               79
Name: distrito, dtype: int64

### Variable Temporalidad

Una vivienda alquilada de forma temporal, debería incroporar en su descripción el concepto de temporal, temporada, con un máximo de 11 meses. Buscamos las viviendas cuya descripción contenga uno de de estos conceptos :

In [40]:
df_Alquiler['temporalidad']= np.where(df_Alquiler['description'].str.contains(r'tempora|Tempora|TEMPORA|11 meses|11 Meses|11 MESES')==True,'Temporal','Indefinido')


In [41]:
df_Alquiler.temporalidad.value_counts()

Indefinido    2642
Temporal      1172
Name: temporalidad, dtype: int64

### Variable cambio_precio

Este concepto se encuentra en la columna 'price' :

In [42]:
df_Alquiler['cambio_precio']= np.where(df_Alquiler['price'].str.contains(r'ha bajado')==True,'Si','No')


In [43]:
df_Alquiler.cambio_precio.value_counts()

No    3087
Si     727
Name: cambio_precio, dtype: int64

### Variable Baños

El número de baños aparece en la columna 'features' :

In [44]:
df_Alquiler['baños_pre']= np.where(df_Alquiler['features'].str.contains(r'habitacion')==True,
                         df_Alquiler['features'].str.split('-').str[2],
                        (np.where(df_Alquiler['features'].str.contains(r'baño')==True,
                         df_Alquiler['features'].str.split('-').str[1],'No')))

In [45]:
df_Alquiler['baños_pre'].value_counts()

 1 baño       1914
 2 baños      1372
 3 baños       293
 4 baños        98
No              74
 5 baños        33
 6 baños         8
 7 baños         7
 9 baños         4
 8 baños         4
 20,73€/m2       1
 1 baño          1
 24,17€/m2       1
 20,77€/m2       1
 16,98€/m2       1
 37,23€/m2       1
 16,67€/m2       1
Name: baños_pre, dtype: int64

Asignamos el número de baños que corresponde, en el caso de que no haya valor, asignamos el valor 1 por defecto akl considerar que una vivienda tiene como mínimo un baño :

In [46]:
df_Alquiler['baños']= np.where(df_Alquiler['baños_pre'].str.contains(r'baño')==True,
                          df_Alquiler['baños_pre'].str.split(' ').str[1],1)

In [47]:
df_Alquiler.baños.value_counts()

1    1915
2    1372
3     293
4      98
1      80
5      33
6       8
7       7
9       4
8       4
Name: baños, dtype: int64

Convertimos la variable a numérica :

In [48]:
df_Alquiler['baños'] = df_Alquiler['baños'].astype('int')

In [49]:
df_Alquiler.drop(['baños_pre'], axis=1, inplace=True)

In [50]:
df_Alquiler.drop(['Unnamed: 0'], axis=1, inplace=True)

### Dataset final

In [51]:
df_Alquiler.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 3814 entries, 0 to 3813
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   id             3814 non-null   object
 1   url            3814 non-null   object
 2   subtype        3814 non-null   object
 3   transaction    3814 non-null   object
 4   owner          3814 non-null   object
 5   sell-type      3814 non-null   object
 6   provider       3814 non-null   object
 7   title          3814 non-null   object
 8   location       3814 non-null   object
 9   features       3814 non-null   object
 10  description    3814 non-null   object
 11  price          3814 non-null   object
 12  price2         3814 non-null   int64 
 13  m2             3814 non-null   int64 
 14  rooms          3814 non-null   int64 
 15  amueblado      3814 non-null   object
 16  parking        3814 non-null   object
 17  distrito       3814 non-null   object
 18  temporalidad   3814 non-null

In [52]:
round(df_Alquiler.describe()).T

count    mean      std  min     25%     50%     75%        max
price2  3814.0  3587.0  47890.0  0.0  1375.0  1892.0  2700.0  2200000.0
m2      3814.0    98.0     74.0  1.0    60.0    80.0   110.0      970.0
rooms   3814.0     2.0      1.0 -1.0     2.0     2.0     3.0       16.0
baños   3814.0     2.0      1.0  1.0     1.0     1.0     2.0        9.0

In [53]:
round(df_Alquiler.describe(include='all')).T

count unique  \
id               3814   3814   
url              3814   3814   
subtype          3814     11   
transaction      3814      1   
owner            3814      2   
sell-type        3814      1   
provider         3814    702   
title            3814   3431   
location         3814     68   
features         3814   3346   
description      3814   3297   
price            3814    959   
price2         3814.0    NaN   
m2             3814.0    NaN   
rooms          3814.0    NaN   
amueblado        3814      2   
parking          3814      2   
distrito         3814     10   
temporalidad     3814      2   
cambio_precio    3814      2   
baños          3814.0    NaN   

                                                                                                                                                                                         top  \
id                                                                                                                                                                           id2277003487234   
url            https://www.habitaclia.com/alquiler-piso-de_100m2_con_dos_habitaciones_totalmente_amueblado_en_plaza_gotic-barcelona-i2277003487234.htm?f=&geo=d&coddists=100&from=list&lo=55   
subtype                                                                                                                                                                                 FLAT   
transaction                                                                                                                                                                             RENT   
owner                                                                                                                                                                           PROFESSIONAL   
sell-type                                                                                                                                                                        SECOND_HAND   
provider                                                                                                                                                                              8166-1   
title                                                                                               Alquiler Piso en Raval. Piso con 2 habitaciones amueblado con ascensor, calefacción y ai   
location                                                                                                                                                     Barcelona - Dreta de l´Eixample   
features                                                                                                                                            50m2 - 1 habitacion - 1 baño - 24,00€/m2   
description                                                                                                      UBK-Disponible 1-11 meses) - Sin posibilidad a visitar! Comprueba la dis...   
price                                                                                                                                                                                2.500 €   
price2                                                                                                                                                                                   NaN   
m2                                                                                                                                                                                       NaN   
rooms                                                                                                                                                                                    NaN   
amueblado                                                                                                                                                                                 Si   
parking                                        

## ANALISIS Y LIMPIEZA DE LA COLUMNAS NUMERICAS

Analizamos los valores máximos y mínimos de las variables numéricas con tal de localizar outliers en el dataset que se correspondan a errores del anuncio :

In [54]:
round(df_Alquiler.describe(),2)

price2       m2    rooms    baños
count     3814.00  3814.00  3814.00  3814.00
mean      3586.61    98.16     2.36     1.66
std      47890.25    73.97     1.31     0.91
min          0.00     1.00    -1.00     1.00
25%       1375.00    60.00     2.00     1.00
50%       1892.50    80.00     2.00     1.00
75%       2700.00   110.00     3.00     2.00
max    2200000.00   970.00    16.00     9.00

### Variable rooms

Analizamos las propiedades con la variable rooms=0

In [55]:
df_Alquiler[(df_Alquiler['rooms']==0)]['subtype'].value_counts()


APARTMENT       17
STUDIO          16
FLAT             9
LOFT             3
PENTHOUSE        2
HOUSE            1
GROUND_FLOOR     1
Name: subtype, dtype: int64

Es posible que una vivienda no tenga ningún dormitorio (studio, loft,etc...) así que aceptamos este dato.

In [56]:
df_Alquiler[(df_Alquiler['rooms']==-1)]['subtype'].value_counts()

FLAT         52
STUDIO       19
LOFT          4
PENTHOUSE     3
APARTMENT     2
DUPLEX        2
Name: subtype, dtype: int64

En cambio no es posible que una vivienda tenga un número negativo de dormitorios, así que eliminamos estos registros:

In [57]:
df_Alquiler.drop(df_Alquiler[df_Alquiler['rooms']==-1].index, inplace = True)

In [58]:
df_Alquiler.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 3732 entries, 0 to 3813
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   id             3732 non-null   object
 1   url            3732 non-null   object
 2   subtype        3732 non-null   object
 3   transaction    3732 non-null   object
 4   owner          3732 non-null   object
 5   sell-type      3732 non-null   object
 6   provider       3732 non-null   object
 7   title          3732 non-null   object
 8   location       3732 non-null   object
 9   features       3732 non-null   object
 10  description    3732 non-null   object
 11  price          3732 non-null   object
 12  price2         3732 non-null   int64 
 13  m2             3732 non-null   int64 
 14  rooms          3732 non-null   int64 
 15  amueblado      3732 non-null   object
 16  parking        3732 non-null   object
 17  distrito       3732 non-null   object
 18  temporalidad   3732 non-null

### Variable precio

Asumimos que el precio máximo de alquiler de una vivienda no puede superar los 30.000€ mensuales, incluso tratando de viviendas de lujo :

In [59]:
df_Alquiler[df_Alquiler['price2']>30000]['subtype'].value_counts()

FLAT    3
LOFT    1
Name: subtype, dtype: int64

In [60]:
df_Alquiler[df_Alquiler['price2']>30000]

id  \
506    id2385003178824   
2397   id2385003181262   
2904  id44731000000006   
3358   id4737003816388   

                                                                                                                                                                                                           url  \
506                                                                      https://www.habitaclia.com/alquiler-loft-con_ascensor_raval-barcelona-i2385003178824.htm?pag=41&f=&geo=d&coddists=100&from=list&lo=55   
2397                                    https://www.habitaclia.com/alquiler-piso-con_5_habitaciones_amueblado_con_ascensor_pedralbes-barcelona-i2385003181262.htm?pag=10&f=&geo=d&coddists=400&from=list&lo=55   
2904  https://www.habitaclia.com/alquiler-piso-pisazo_de_3_habitaciones_en_el_corazon_de_glorias_el_parc_i_la_llacuna_del_poblenou-barcelona-i44731000000006.htm?pag=24&f=&geo=d&coddists=1000&from=list&lo=55   
3358                      https://www.habitaclia.com/alquiler-piso-con_dos_plazas_de_parking_en_venta_en_les_tres_torres_tres_torres-barcelona-i4737003816388.htm?pag=13&f=&geo=d&coddists=500&from=list&lo=55   

     subtype transaction         owner    sell-type provider  \
506     LOFT        RENT  PROFESSIONAL  SECOND_HAND   2385-1   
2397    FLAT        RENT  PROFESSIONAL  SECOND_HAND   2385-1   
2904    FLAT        RENT  PROFESSIONAL  SECOND_HAND  44731-1   
3358    FLAT        RENT  PROFESSIONAL  SECOND_HAND   4737-1   

                                                                                                       title  \
506                                                                Alquiler Loft en Raval. Loft con ascensor   
2397                              Alquiler Piso en Pedralbes. Piso con 5 habitaciones amueblado con ascensor   
2904  Alquiler Piso en El Parc i la Llacuna del Poblenou. Pisazo de 3 habitaciones, en el corazón de glorias   
3358             Alquiler Piso Carrer del rosari. Piso con dos plazas de parking en venta en les tres torres   

                                           location  \
506                               Barcelona - Raval   
2397                          Barcelona - Pedralbes   
2904  Barcelona - El Parc i la Llacuna del Poblenou   
3358                        Barcelona - Tres Torres   

                                             features  \
506       35m2 - 1 habitacion - 1 baño - 4.714,29€/m2   
2397  330m2 - 5 habitaciones - 4 baños - 5.909,09€/m2   
2904    68m2 - 3 habitaciones - 1 baño - 4.375,00€/m2   
3358  283m2 - 5 habitaciones - 4 baños - 7.773,85€/m2   

                                                                                                                      description  \
506             Ref: 92057 Magnifico Loft moderno de 35 m², situado en la 3ºplanta con ascensor Buen estado de conservación La...   
2397        REF 93190 Casa en venta con superficie de 330m², en la zona de Les Corts, cerca del Camp Nou. Exterior con piscina...   
2904  En el Corazón de Glories, un barrio en plena ebullición y rodeado de obra nueva encontramos este magnifica entresuelo de...   
3358  Precioso piso exterior muy luminoso en Tres Torres de casi 300 m2 (284m2 construidos de vivienda + 16m2 terraza principa...   

            price   price2   m2  rooms amueblado parking             distrito  \
506     165.000 €   165000   35      1        No      No         Ciutat Vella   
2397  1.950.000 €  1950000  330      5        Si      No            Les Corts   
2904    297.500 €   297500   68      3        No      No           Sant Marti   
3358  2.200.000 €  2200000  283      5        Si      Si  Sarria Sant Gervasi   

     temporalidad cambio_precio  baños  
506    Indefinido            No      1  
2397   Indefinido            No      4  
2904   Indefinido            No      1  
3358   Indefinido            No      4

In [61]:
df_Alquiler.drop(df_Alquiler[df_Alquiler['price2']>30000].index, inplace = True)

No puede ser tampoco el precio de alquiler sea nulo o inferior a 300€ al mes :

In [62]:
df_Alquiler[df_Alquiler['price2']<300]['subtype'].value_counts()

APARTMENT    1
Name: subtype, dtype: int64

In [63]:
df_Alquiler[df_Alquiler['price2']<300]

id  \
752  id30785000001911   

                                                                                                                                                                                       url  \
752  https://www.habitaclia.com/alquiler-apartamento-amueblado_con_calefaccion_y_aire_acondicionado_barceloneta-barcelona-i30785000001911.htm?pag=61&f=&geo=d&coddists=100&from=list&lo=55   

       subtype transaction         owner    sell-type provider  \
752  APARTMENT        RENT  PROFESSIONAL  SECOND_HAND  30785-1   

                                                                                                title  \
752  Alquiler Apartamento Calle atlántida. Apartamento amueblado con calefacción y aire acondicionado   

                    location                      features  \
752  Barcelona - Barceloneta  38m2 - 1 habitacion - 1 baño   

                                                                                                             description  \
752  CORTO PLAZO;  JULIO Y AGOSTO DE 2023 // LARGO PLAZO ; DE OCTUBRE DE 2023 ALQUILER TEMPORAL 1 A 11 MESES Bienveni...   

           price  price2  m2  rooms amueblado parking      distrito  \
752  Oportunidad       0  38      1        Si      No  Ciutat Vella   

    temporalidad cambio_precio  baños  
752     Temporal            No      1

In [64]:
df_Alquiler.drop(df_Alquiler[df_Alquiler['price2']<300].index, inplace = True)

### Variable m2

In [65]:
df_Alquiler[df_Alquiler['m2']<20]

id  \
1759   id4807004429387   
3157  id30028000217870   

                                                                                                                                                                             url  \
1759       https://www.habitaclia.com/alquiler-piso-con_2_habitaciones_l_antiga_esquerra_de_l_eixample-barcelona-i4807004429387.htm?pag=74&f=&geo=d&coddists=200&from=list&lo=55   
3157  https://www.habitaclia.com/alquiler-piso-en_en_41742_font_honrada_el_poble_sec_parc_de_montjuic-barcelona-i30028000217870.htm?pag=17&f=&geo=d&coddists=300&from=list&lo=55   

     subtype transaction         owner    sell-type provider  \
1759    FLAT        RENT  PROFESSIONAL  SECOND_HAND   4807-1   
3157    FLAT        RENT  PROFESSIONAL  SECOND_HAND  30028-1   

                                                                                                                title  \
1759                                                          Alquiler Piso Passatge d'aragó. Piso con 2 habitaciones   
3157  Alquiler Piso en El Poble Sec - Parc de Montjuïc. Piso en alquiler en 41742, barcelona (barcelona) font honrada   

                                         location  \
1759  Barcelona - L´Antiga Esquerra de l´Eixample   
3157  Barcelona - El Poble Sec - Parc de Montjuïc   

                                          features  \
1759    8m2 - 2 habitaciones - 1 baño - 156,25€/m2   
3157  1m2 - 2 habitaciones - 1 baño - 1.350,00€/m2   

                                                                                                                      description  \
1759  Vivienda situada en el barrio del Eixample esquerra, dispone de 80m2, distribuidos en 2 amplias habitaciones, una de ell...   
3157                                                                             Piso en Poble Sec de 2 habitaciones y balcón....   

                        price  price2  m2  rooms amueblado parking  \
1759  1.250 €\nha bajado 50 €    1250   8      2        No      No   
3157                  1.350 €    1350   1      2        No      No   

            distrito temporalidad cambio_precio  baños  
1759        Eixample   Indefinido            Si      1  
3157  Sants Montjuic   Indefinido            No      1

In [66]:
df_Alquiler.drop(df_Alquiler[df_Alquiler['m2']<20].index, inplace = True)

In [67]:
df_Alquiler.describe()

price2           m2        rooms        baños
count   3725.000000  3725.000000  3725.000000  3725.000000
mean    2392.035973    98.616107     2.436242     1.675436
std     1945.115674    73.336549     1.221244     0.917458
min      600.000000    20.000000     0.000000     1.000000
25%     1390.000000    60.000000     2.000000     1.000000
50%     1895.000000    80.000000     2.000000     1.000000
75%     2700.000000   110.000000     3.000000     2.000000
max    25000.000000   970.000000    16.000000     9.000000

### Variable baños

In [68]:
df_Alquiler[df_Alquiler['baños']>5]['subtype'].value_counts()

HOUSE        15
FLAT          5
DUPLEX        2
PENTHOUSE     1
Name: subtype, dtype: int64

In [69]:
df_Alquiler.to_csv('df_Alquiler_final.csv')